### PR TITLE
Implement task editor in survey editor flow

### PR DESCRIPTION
**Pull request by os-micmec**
Towards #1053

### CODE REVIEW

File: web/src/app/pages/create-survey/task-details/task-input/task-input.component.html [hunk 1]
```diff
@@ -115,5 +118,15 @@
         Add option
       </button>
     </div>
+<<<<<<< HEAD
```
Comment (gino-m) [inline web/src/app/pages/create-survey/task-details/task-input/task-input.component.html:121]: Merge conflict

File: web/src/app/pages/edit-survey/edit-job/edit-job.component.ts [hunk 2]
```diff
@@ -14,20 +14,155 @@
...
+    );
+
+    this.tasks = this.tasks!.sortBy(task => task.index);
+  }
+
+  getIndex(index: number) {
+    return index;
+  }
+
+  onTaskAdd(group: TaskGroup) {
```
Comment (gino-m) [inline web/src/app/pages/edit-survey/edit-job/edit-job.component.ts:88]: Nit: onAddTask

File: web/src/app/pages/edit-survey/edit-job/edit-job.component.ts [hunk 3]
```diff
@@ -14,20 +14,155 @@
...
+      );
+
+      this.tasks = this.tasks.push(task);
+    }
+
+    // No need to call addOrUpdateTasks because ngOnChange emits an update once
+    // the task-input component is created.
+  }
+
+  onTaskUpdate(event: Task, index: number) {
```
Comment (gino-m) [inline web/src/app/pages/edit-survey/edit-job/edit-job.component.ts:108]: Nit: onUpdateTask

File: web/src/app/pages/edit-survey/edit-job/edit-job.component.ts [hunk 4]
```diff
@@ -14,20 +14,155 @@
...
+    if (!this.tasks) {
+      throw Error('tasks list is is empty');
+    }
+
+    this.tasks = this.tasks.set(index, event);
+    this.taskService.addOrUpdateTasks(this.surveyId!, this.jobId!, this.tasks);
+  }
+
+  onTaskDelete(index: number) {
```
Comment (gino-m) [inline web/src/app/pages/edit-survey/edit-job/edit-job.component.ts:117]: Nit: onDeleteTask

File: web/src/app/pages/edit-survey/edit-job/edit-job.component.ts (hunk 3 above)
Comment (os-micmec) [reply web/src/app/pages/edit-survey/edit-job/edit-job.component.ts:108]: Done

File: web/src/app/pages/edit-survey/edit-job/edit-job.component.ts (hunk 2 above)
Comment (os-micmec) [reply web/src/app/pages/edit-survey/edit-job/edit-job.component.ts:88]: Done

File: web/src/app/pages/edit-survey/edit-job/edit-job.component.ts (hunk 4 above)
Comment (os-micmec) [reply web/src/app/pages/edit-survey/edit-job/edit-job.component.ts:117]: Done

**Review by rfontanarosa — APPROVED** [review]

**Pull request merged**